core/impl/src/main/java/org/eclipse/sensinact/core/whiteboard/impl/SensinactWhiteboard.java

```diff
@@ -133,6 +135,48 @@ public SensinactWhiteboard(GatewayThread gatewayThread, IMetricsManager metrics)
         this.metrics = metrics;
     }
 
+    private Map<RegistryKey, WhiteboardContext<WhiteboardGetSet<?>>> rcHandlers = new HashMap<>();
```

Comment:
Would it make more sense to introduce these as `ActMethod`, `GetMethod` and `SetMethod` types so that we only have one registry to check each time we do anything?

Reply:
On reflection I think that there might be some benefit in extracting interfaces for `GetMethod`, `SetMethod` and `ActMethod` which we use internally to abstract between annotated and non-annotated whiteboard services, the existing implementations then work for annotated methods and new (simple) implementations work for whiteboard services.

We could then use the same updated method for all types in the SensinactWhitebaord

Reply:
That was the initial plan, but returned types are not exactly the same (e.g. value instead of promise of value for actions)

Reply:
The return is deliberately fuzzy for annotated, it can be a promise or not: 

https://github.com/eclipse/org.eclipse.sensinact.gateway/blob/fefa2b29b52e2339061cd82f296a173818365d7d/core/impl/src/main/java/org/eclipse/sensinact/core/whiteboard/impl/SensinactWhiteboard.java#L506-L510 

The interface versions will just be less fuzzy